System/Directory/Internal/Posix.hsc

```diff
@@ -348,22 +354,46 @@ getHomeDirectoryInternal = do
            (Posix.getEffectiveUserID >>= Posix.getUserEntryForID)
 
 getXdgDirectoryFallback :: IO OsPath -> XdgDirectory -> IO OsPath
+#ifdef haiku_HOST_OS
```

Comment:
This code is supposed to mirror the [specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), which does not have Haiku-specific exceptions.

Reply:
Haiku is not a Linux desktop, it's a BeOS clone, doesn't have folders like /usr, doesn't use X11/Wayland (which the XDG spec is for), nor does it use dotfiles. And given that GHC directly depends on this library, and there's only been a one-line change to a patch I've been carrying since GHC 8.4 that was trivial; and I'm happy to help maintain going forward.